src/store/index.ts

```diff
@@ -0,0 +1,116 @@
+import { createStore } from 'vuex'
+import VuexPersist from 'vuex-persist'
+import State, { Pokemon, Team } from '../interfaces'
+import ditto from './tempPoke'
+import { idGenerator } from '../Common/idGenerator'
+import Http from '../services/Api'
+
+const localStorageData = JSON.parse(<any>localStorage.getItem('vuex'))
+const generateId = idGenerator(localStorageData?.teamsList.reduce((acum: number, team:Team)=> {
```

Comment:
See if there is something in `lodash` that helps cleaning this up: https://lodash.com